apps/coordinator/src/utils/psbtUtils.ts

```diff
@@ -0,0 +1,309 @@
+  addressUsed: boolean;
+  addressKnown: boolean;
+}
+
+/**
+ * Interface for slice data structure
+ */
+interface Slice {
+  multisig: any; // Type this more specifically based on our multisig structure
+  bip32Path: string;
+  change: boolean;
+}
+
+/**
+ * Interface for input objects with multisig information
+ */
+interface Input extends UTXO {
+  multisig: Slice["multisig"]; // From slice.multisig
+  bip32Path: Slice["bip32Path"]; // From slice.bip32Path
+}
+
+/**
+ * Interface for signature information extracted from PSBT
+ */
+interface SignatureInfo {
+  signature: Buffer | string;
+  pubkey: string;
+  inputIndex: number;
+}
+
+/**
+ * Interface for signature sets grouped by signer
+ */
+interface SignatureSet {
+  signatures: Buffer[] | string[];
+  publicKeys: string[];
+  signerPubkey: string;
+}
+
+/**
+ * Loads a PSBT from a string or buffer, handling both PSBTv0 and PSBTv2 formats.
+ * Uses only functionality from the @caravan/psbt package removing dependency on the outdated use of @caravan/bitcoin
+ *
+ * @param psbtText - The PSBT as a string or buffer
+ * @param network - The Bitcoin network (mainnet, testnet, etc.)
+ * @returns The PSBT object compatible with bitcoinjs-lib
+ * @throws Error if the PSBT cannot be parsed
+ */
+export function loadPsbt(
+  psbtText: string | Buffer,
+  network: Network,
+): Psbt | null {
+  const options = { network: networkData(network) };
+
+  try {
+    // Try to determine the PSBT version
+    const version = getPsbtVersionNumber(psbtText);
+
+    if (version >= 2) {
+      // It's a PSBTv2, convert it to PSBTv0 format for compatibility
+      console.log(
+        `Detected PSBTv2 (version ${version}), converting to PSBTv0 format`,
+      );
+      try {
+        const psbtv2 = new PsbtV2(psbtText, true); // Allow version 1 transactions if needed
+        const psbtv0Base64 = psbtv2.toV0("base64");
+        // Use the autoLoadPSBT from @caravan/psbt
+        return psbtPackageAutoLoad(psbtv0Base64, options);
+      } catch (e) {
+        console.error("Failed to parse PSBTv2:", e);
+        throw new Error(`Failed to parse PSBTv2: ${e.message}`);
+      }
+    } else {
+      // It's a PSBTv0, use the autoLoadPSBT from @caravan/psbt
+      const psbt = psbtPackageAutoLoad(psbtText, options);
+      if (!psbt) {
+        throw new Error("Could not parse PSBT. Invalid format.");
+      }
+      return psbt;
+    }
+  } catch (e) {
+    console.error("Error loading PSBT:", e);
+    throw new Error(`Error loading PSBT: ${e.message}`);
+  }
+}
+
+/**
+ * Extracts signatures from a PSBT and organizes them by signer.
+ *
+ * @param  psbt - The PSBT object
+ * @param  inputs - Array of input objects with multisig info
+ * @param  outputs - Array of output objects
+ * @param  network - Network (mainnet, testnet, etc.)
+ * @returns  Array of signature sets, each containing signatures for all inputs
+ */
+export function extractSignaturesFromPSBT(psbt: Psbt, inputs: Input[]) {
```

Comment:
noted in the other file, but this should probably be a selector that way you can pull the inputs from the psbt using the redux store.